### ISSUE TITLE
`reportUninitializedInstanceVariable` should detect when the attribute is initialized in `__new__`

### ISSUE THREAD

**Issue by DetachHead** [DetachHead/basedpyright#790]
Code sample in [basedpyright playground](https://basedpyright.com/?pythonVersion=3.9&typeCheckingMode=all&reportUnusedExpression=false&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoDKApgDbACwAUFVQMYkCGAzk1AGJhgBcVUfUwTl0wp8AYihEQ4EMJBEceAKopUSGEgYkkALyIATAJIomMBilpEAagxCaARiSI1K-KPqLAoAfW8oiAO6%2BABT0TACUUAC0AHyEpMA8rm78TAlQALxQTACuCFLB4QB0vv5B3qEkEbwpfGlkRYJgmVAAjDW18jA5ICjZCUA)

```python
from typing import Self


class Foo:
    foo: int # error: reportUninitializedInstanceVariable

    def __new__(cls) -> Self:
        self = super().__new__(cls)
        self.foo = 1
        return self
```

**Comment by DetachHead** [comment]
Duplicate of #455
